vms/platformvm/txs/executor/backend.go

```diff
@@ -23,4 +24,5 @@ type Backend struct {
 	Uptimes      uptime.Manager
 	Rewards      reward.Calculator
 	Bootstrapped *utils.Atomic[bool]
+	Mempool      mempool.Mempool
```

Comment:
I don't think we need mempool to be an attribute of the backend. Just drop it and simply pass it in blockbuilder.New

Reply:
We already both use mempool + txExecutorBackend as arguments in `blockbuilder.New` and `blockexecutor.NewManager`. I thought `Backend` serves as provider for both of these `New` functions, so that we can use `Backend` to group common objects. (instead of passing it as separate argument)

Reply:
I see but the backend is used in other places where mempool is not needed. Not sure if I am changing my mind here, but now it feels cleaner to keep mempool our of backend.
Anyhow, we can let other reviewers express their opinion on this